kotest-tests/kotest-tests-spec-parallelism/src/jvmTest/kotlin/com/sksamuel/kotest/parallelism/ProjectConfig.kt

```diff
@@ -17,10 +23,11 @@ object ProjectConfig : AbstractProjectConfig() {
    override val concurrentSpecs: Int = ProjectConfiguration.MaxConcurrency
 
    override suspend fun afterProject() {
-      val duration = System.currentTimeMillis() - start
-      // there are 8 specs, and each one has a delay
-      // if parallel is working they should all block at the same time
-      if (duration > 700)
+      val duration = start.elapsedNow()
+      // There are 8 specs, and each one has a 1-second delay.
+      // If parallel is working, they should all block at the same time.
+      if (duration > 7.seconds) {
```

Comment:
If the tests are at 100 ms each, total sequential execution time would be 800 ms so this could almost never fail.

Maybe a better approach would be to increase the blocking time inside the test threads from `Thread.sleep(100)` to `Thread.sleep(400)` to handle a slow CI and then check like `duration > (7*400).milliseconds`?

Reply:
Oooooops yes, I misread 100 as 1000! I'll fix it.


Your proposal may well work, but just to note, I don't think the speed of the CI is the problem. When I ran `./gradlew jvmTest --rerun-tasks` I saw the same problem using my Macbook.

I'll fix the times and play around some more...

Reply:
Weird things happen in real time. 🙃 Usually my machine is the slowest one around but it ran the 8 tests in 170 ms so well below the former 700 ms limit.

Reply:
I've bumped the blocking time - let's see if it helps!

Alternatively, it would probably work if `Thread.sleep()` was replaced with a coroutines `delay()` - but I don't know if using `Thread.sleep()` is the point of the test.

Reply:
In this case I'd guess it makes no difference: Each test runs on its own thread via its own dispatcher. Since that dispatcher has no other jobs to schedule in case of a `delay()`, it would most probably do the same thing, i.e. invoke `Thread.sleep()`.

Reply:
Nice!

Good point about each test having its own dispatcher. Maybe telling Kotest to use  [TestCoroutineScheduler](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/) would work? Then measure the virtual time, not the actual elapsed time.

Reply:
> Executed 8 tests in 7s

Why is it 7 seconds and not 3.5 seconds? 🤔

Reply:
> Maybe telling Kotest to use TestCoroutineScheduler would work? Then measure the virtual time, not the actual elapsed time.
That would defeat the purpose: Each test would complete immediately. We'd have no way to detect actual cross-test parallelism. 

 > Why is it 7 seconds and not 3.5 seconds?
 The actual execution time to expect would be slightly above 0.5 seconds (all 8 tests in parallel sleeping for 500 ms at the same time). The rest is just effects of CI system load (or artificial slowdown, I don'nt know if they do this).